### ISSUE TITLE
Tensorflow 102 강좌 중 Conv2D에서 채널

### ISSUE THREAD

**Issue by sohicode** [codingeverybody/codingyahac#1094]
# 해결하고자 하는 문제
Tensorflow 102 강좌 중 CNN의 Conv2D 동영상에서 나오는 코드의 Convolution 채널 및 파라미터 계산 문제

# CNN모델을 만들기 코드
X = tf.keras.layers.Input(shape=[28, 28, 1])
H = tf.keras.layers.Conv2D(3, kernel_size=5, activation='swish')(X)
H = tf.keras.layers.Conv2D(6, kernel_size=5, activation='swish')(H)
H = tf.keras.layers.Flatten()(H)
H = tf.keras.layers.Dense(84, activation='swish')(H)
Y = tf.keras.layers.Dense(10, activation='softmax')(H)
model = tf.keras.models.Model(X, Y)
model.compile(loss='categorical_crossentropy', metrics='accuracy')

# 질문
동영상 강좌 다음 부분: https://youtu.be/5lov3_2qkK4?t=313

Conv2D(3, kernel_size=5, activation='swish') -->  filter가 (5,5,1)x3
계산식이 filter size인 5,5와 input의 채널인 1이라서 (5,5,1)이고 output 채널이 3이라서 3을 곱한 것인 것 같고,

Conv2D(6, kernel_size=5, activation='swish') -->  filter가 (5,5,3)x6
계산식이 filter size인 5,5와 input의 채널인 3이라서 (5,5,3)이고 output 채널이 6이라서 6을 곱한 것인 것 같은데

이 부분이 왜 이렇게 계산하는 것인지 잘 이해가 안됩니다.
Conv2D의 함수 내용에 대한 이해가 부족해서 그런 것 같습니다. 더 자세한 설명 부탁드려요 *^^*

**Comment by storykim** [comment]
안녕하세요.

쉬운 예시부터 시작해보겠습니다.
kernel size가 5, input 채널이 3, output 채널이 1인 경우를 생각해보면,
filter는 (5, 5, 3) x 1이 됩니다.

여기서 (5, 5, 3)이 나온 이유는 5x5 크기의 필터(커널)를 3개의 인풋 채널에 적용시켜서 계산을 하기 위함이고,
마지막에 x1을 해준 이유는 계산을 통해 하나의 값(output 채널 = 1)만 얻으면 되기 때문입니다.

그럼 만약 output 채널이 6이면 어떻게 될까요?
(5, 5, 3) x 6이 됩니다. 그 이유는, 5x5 크기의 필터를 3개의 인풋 채널에 적용시켜서 계산을 하는데, 그런 계산을 통해 6개의 다른 값을 얻고싶기 때문에 필터가 6개 필요한 것입니다.

혹시 설명이 조금 더 필요한 부분이 있으시다면 말씀해주세요!

**Comment by sohicode** [comment]
그럼 첫번째는 (5,5,1)x3이고,, 여기서 나온 output채널이 3이니깐
다음 (5,5,3)x6 에서 input이 3이 된 것이군요. 그리고 output이 6인거고요. 
이해되었습니다. 잘 설명해주셔서 감사합니다.*^^*

그런데,, 이슈가 해결되었으면 아래 [Close issue] 버튼을 누르는 것인가요? 아니면 가만 둬야하나요?

**Comment by storykim** [comment]
Close issue 버튼을 누르면 됩니다. 이번엔 제가 누를게요 :)
